packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -647,7 +647,7 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
       if (
         await observer.didPushRouteInformation(
           RouteInformation(
-            location: routeArguments['location'] as String,
+            uri: Uri.parse(routeArguments['location'] as String),
```

Comment:
What is the purpose of this PR? It seems like all it effectively does is parse the string here into an URI, only to convert it back to a string above in line 91. It never does anything useful with the URI representation?

(Or is this just meant as a check that `location` has proper URI format? In that case, should we just do the parsing in an assert? Or is this user-provided input and we need to always verify this even in release mode?)

Reply:
The purpose is to prepare this api so that engine can start sending full uri like `http://www.mywebsite.com/mypath`. This is important for apps that want to handle deeplinks from different web domains. Right now if we do it directly this will be a breaking change.

The line 91 is for backward compatibility that does not use router API.

Reply:
This pr also add new getter `RouteInformation.uri`.

It lets developer to query RouteInformation.uri.scheme or RouteInformation.uri.host when they write their routing code.